This patch is based on the commit described below, taken from the upstream
time64 pull request and adapted for use in the Debian package by
Peter Michael Green.

commit c2f9db639c38f34bda0d999d557f5cec0fb80915
Author: Ola x Nilsson <olani@axis.com>
Date:   Mon Mar 20 14:22:31 2023 +0100

    gnu: Update struct shmid_ds for 64-bit time

Index: b/src/unix/linux_like/linux/gnu/b32/arm/mod.rs
===================================================================
--- a/src/unix/linux_like/linux/gnu/b32/arm/mod.rs
+++ b/src/unix/linux_like/linux/gnu/b32/arm/mod.rs
@@ -116,10 +116,13 @@ s! {
         pub shm_perm: crate::ipc_perm,
         pub shm_segsz: size_t,
         pub shm_atime: crate::time_t,
+        #[cfg(not(gnu_time64_abi))]
         __unused1: c_ulong,
         pub shm_dtime: crate::time_t,
+        #[cfg(not(gnu_time64_abi))]
         __unused2: c_ulong,
         pub shm_ctime: crate::time_t,
+        #[cfg(not(gnu_time64_abi))]
         __unused3: c_ulong,
         pub shm_cpid: crate::pid_t,
         pub shm_lpid: crate::pid_t,
Index: b/src/unix/linux_like/linux/gnu/b32/powerpc.rs
===================================================================
--- a/src/unix/linux_like/linux/gnu/b32/powerpc.rs
+++ b/src/unix/linux_like/linux/gnu/b32/powerpc.rs
@@ -113,14 +113,10 @@ s! {
 
     pub struct shmid_ds {
         pub shm_perm: crate::ipc_perm,
-        __glibc_reserved1: c_uint,
+        pub shm_segsz: crate::size_t,
         pub shm_atime: crate::time_t,
-        __glibc_reserved2: c_uint,
         pub shm_dtime: crate::time_t,
-        __glibc_reserved3: c_uint,
         pub shm_ctime: crate::time_t,
-        __glibc_reserved4: c_uint,
-        pub shm_segsz: size_t,
         pub shm_cpid: crate::pid_t,
         pub shm_lpid: crate::pid_t,
         pub shm_nattch: crate::shmatt_t,
Index: b/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
===================================================================
--- a/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
+++ b/src/unix/linux_like/linux/gnu/b32/x86/mod.rs
@@ -190,11 +190,8 @@ s! {
         pub shm_perm: crate::ipc_perm,
         pub shm_segsz: size_t,
         pub shm_atime: crate::time_t,
-        __unused1: c_ulong,
         pub shm_dtime: crate::time_t,
-        __unused2: c_ulong,
         pub shm_ctime: crate::time_t,
-        __unused3: c_ulong,
         pub shm_cpid: crate::pid_t,
         pub shm_lpid: crate::pid_t,
         pub shm_nattch: crate::shmatt_t,
